File: packages/pensopay_android/android/src/main/kotlin/com/flutter/pensopay/networking/pensopayapi/pensopaylink/mandate/PPGetMandateRequest.kt
package com.flutter.pensopay.networking.pensopayapi.pensopaylink.mandate

import com.android.volley.Request
import com.flutter.pensopay.networking.pensopayapi.pensopaylink.models.PPMandate
import com.flutter.pensopay.networking.pensopayapi.PPrequest

class PPGetMandateRequest(subscription_id: Int, mandate_id: String): PPrequest<PPMandate>(Request.Method.GET, "/subscription/$subscription_id/mandate/$mandate_id", null, PPMandate::class.java)

File: packages/pensopay_ios/ios/Classes/Networking/Pensopay API/Pensopay Link/Mandates/PPGetMandateRequest.swift
//
//  PPGetMandateRequest.swift
//  PensopaySDK
//
//  Created on 07/28/2022
//  Copyright © 2022 Pensopay. All rights reserved.
//

import Foundation

public class PPGetMandateRequest : PPRequest {
    
    // MARK: - Properties
    
    public var subscription_id: Int
    public var mandate_id: String

    
    // MARK: - Init
    
    public init(subscription_id: Int, mandate_id: String) {
        self.subscription_id = subscription_id
        self.mandate_id = mandate_id
    }
    
    
    // MARK: - URL Request
    
    public func sendRequest(success: @escaping (_ result: PPMandate) -> Void, failure: ((_ data: Data?, _ response: URLResponse?, _ error: Error?) -> Void)?) {
        guard let url = URL(string: "\(PensopayAPIBaseUrl)/subscription/\(self.subscription_id)/mandate/\(self.mandate_id)") else {
            return
        }
        
        var request = URLRequest(url: url)
        request.httpMethod = "GET"
        request.setValue(headers.encodedAuthorization(), forHTTPHeaderField: "Authorization")
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        request.setValue("application/json", forHTTPHeaderField: "Accept")
        
        super.sendRequest(request: request, success: success, failure: failure)
    }
}
